commit 70ab2cd108663a1c9c31dd17a43dd58c453fe900
Author: Christoph Moench-Tegeder <cmt@FreeBSD.org>
Date:   Fri Feb 20 21:15:44 2026 +0100

    powerpc64le support for FreeBSD
    
    Adapted from:
      commit 192a67a0d79f89529eaac61ea9e865b7cb32365e
      Author: Piotr Kubaj <pkubaj@FreeBSD.org>
      Date:   Sat May 15 02:45:46 2021 +0000
    
        security/nss: fix build on powerpc64le

diff --git lib/freebl/Makefile lib/freebl/Makefile
index 9286e436f..3509a0656 100644
--- lib/freebl/Makefile
+++ lib/freebl/Makefile
@@ -263,7 +263,7 @@ else ifeq ($(CPU_ARCH),x86)
 endif
 endif # Darwin
 
-ifeq ($(OS_TARGET),Linux)
+ifeq (,$(filter-out Linux FreeBSD, $(OS_TARGET)))
 ifeq ($(CPU_ARCH),x86_64)
     # Lower case s on mpi_amd64_common due to make implicit rules.
     ASFILES  = arcfour-amd64-gas.s mpi_amd64_common.s
@@ -297,11 +297,12 @@ ifdef USE_64
     PPC_ABI := $(shell $(CC) -dM -E - < /dev/null | awk '$$2 == "_CALL_ELF" {print $$3}')
     ifeq ($(PPC_ABI),2)
         ASFILES += sha512-p8.s
-    ifeq ($(OS_TEST),ppc64le)
+    ifeq (,$(filter-out powerpc64le ppc64le, $(OS_TEST)))
         EXTRA_SRCS += chacha20poly1305-ppc.c ppc-gcm-wrap.c
         ASFILES += chacha20-ppc64le.s ppc-gcm.s
         DEFINES += -DHAVE_PLATFORM_GCM
-    endif # ppc64le
+        ASFLAGS += -fno-integrated-as
+    endif # powerpc64le, ppc64le
     endif
 endif # USE_64
 endif # ppc
